girder/web/client/package.json

```diff
@@ -11,8 +11,6 @@
   },
   "dependencies": {
     "@girder/core": "file:../core",
-    "@girder/girder-plugin-homepage": "file:../../../plugins/homepage/girder_homepage/web_client",
-    "@girder/girder-plugin-jobs": "file:../../../plugins/jobs/girder_jobs/web_client",
```

Comment:
This is a key change: the `client` package must avoid having to statically include all plugin dependencies, in order to support plugins outside of the `girder/girder` repository.

Reply:
@zachmullen Ah, if I understand it well enough, this is clever. By adding the API endpoint to retrieve plugin source URLs, you are essentially moving plugin code inclusion from being a build time concern to being a runtime concern, obviating the need for updating the `package.json` for a client build for each package.

Essentially this is taking the "remove webroot.mako" that I had done to its logical conclusion. I had used JS and the build system to include (topo-sorted) assets, but yours method is cleaner. Just like I moved getting settings from mako to an endpoint queried at runtime, you also move plugin inclusion (which had been in mako), to an endpoint. This is clean and consistent IMO.

Also I like that there's an explicit list of objects to include in the frontend instead of using glob expressions for js, css.